### PR TITLE
Run etcd for events on separate machine in Kubemark

### PR DESCRIPTION
**Pull request by gmarek**
cc @quinton-hoole

### CODE REVIEW

**Comment by gmarek** [comment]
cc @wojtek-t

File: test/kubemark/start-kubemark-master.sh [hunk 1]
```diff
@@ -15,18 +15,25 @@
 # limitations under the License.
 
 # TODO: figure out how to get etcd tag from some real configuration and put it here.
+
+EVENT_STORE_IP=$(cat /tmp/event-store-ip)
```
Comment (quinton-hoole) [inline test/kubemark/start-kubemark-master.sh]: Can we avoid passing variables around via tmp files?

File: test/kubemark/start-kubemark-master.sh [hunk 2]
```diff
@@ -15,18 +15,25 @@
...
+if [ "${EVENT_STORE_IP}" == "127.0.0.1" ]; then
```
Comment (quinton-hoole) [inline test/kubemark/start-kubemark-master.sh]: Do we really need to special case localhost like this? Can we not run it on port 4002 irrespective of whether it's local or not?

File: test/kubemark/start-kubemark.sh [hunk 3]
```diff
@@ -97,7 +138,8 @@ gcloud compute ssh --zone=${ZONE} --project="${PROJECT}" ${MASTER_NAME} \
   sudo bash -c \"echo \"${KUBE_BEARER_TOKEN},admin,admin\" > /srv/kubernetes/known_tokens.csv\" && \
   sudo bash -c \"echo \"${KUBELET_TOKEN},kubelet,kubelet\" >> /srv/kubernetes/known_tokens.csv\" && \
   sudo bash -c \"echo \"${KUBE_PROXY_TOKEN},kube_proxy,kube_proxy\" >> /srv/kubernetes/known_tokens.csv\" && \
-  sudo bash -c \"echo admin,admin,admin > /srv/kubernetes/basic_auth.csv\""
+  sudo bash -c \"echo admin,admin,admin > /srv/kubernetes/basic_auth.csv\" && \
+  sudo bash -c \"echo -n ${EVENT_STORE_IP:-127.0.0.1} > /tmp/event-store-ip\""
```
Comment (quinton-hoole) [inline test/kubemark/start-kubemark.sh]: Comment as per above.  Can we avoid passing variables around via tmp files?

File: test/kubemark/stop-kubemark.sh [hunk 4]
```diff
@@ -24,6 +24,7 @@ source "${KUBE_ROOT}/cluster/kube-env.sh"
 detect-project &> /dev/null
 
 MASTER_NAME="${INSTANCE_PREFIX}-kubemark-master"
+EVENT_STORE_NAME="${INSTANCE_PREFIX}-event-store"
```
Comment (quinton-hoole) [inline test/kubemark/stop-kubemark.sh]: This is a duplicate of what's in start-kubelet.sh.  Can we factor it out into a common place?

File: test/kubemark/start-kubemark.sh [hunk 5]
```diff
@@ -68,11 +68,52 @@ gcloud compute firewall-rules create "${INSTANCE_PREFIX}-kubemark-master-https"
...
-    --allow "tcp:443" || true
+    --allow "tcp:443"
 
 MASTER_IP=$(gcloud compute instances describe ${MASTER_NAME} \
   --zone="${ZONE}" --project="${PROJECT}" | grep natIP: | cut -f2 -d":" | sed "s/ //g")
 
+if [ "${SEPARATE_EVENT_MACHINE:-false}" == "true" ]; then
+  EVENT_STORE_NAME="${INSTANCE_PREFIX}-event-store"
+  gcloud compute disks create "${EVENT_STORE_NAME}-pd" \
+      --project "${PROJECT}" \
```
Comment (quinton-hoole) [inline test/kubemark/start-kubemark.sh]: Can we factor out all of the common gcloud command line arguments into one central place, perhaps something like $GCLOUD_COMMON_ARGS (e.g. --project, --zone, --quiet)

**Comment by quinton-hoole** [comment]
LGTM barring relatively minor comments, which are mainly around factoring out cargo cult code.

File: test/kubemark/start-kubemark-master.sh (hunk 1 above)
Comment (gmarek) [inline test/kubemark/start-kubemark-master.sh]: Well, I can put it in the .bashrc (or something), but I'm not sure it is any better. And I'd rather avoid using salt.

File: test/kubemark/start-kubemark-master.sh (hunk 2 above)
Comment (gmarek) [inline test/kubemark/start-kubemark-master.sh]: Yup.

File: test/kubemark/start-kubemark.sh (hunk 3 above)
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: I don't have any better idea. I can either set it directly in .bashrc-ish thing, use salt (or sth), or pass it directly to the bash command running start-kubemark-master.sh (which makes it more unreadable in my opinion).

If you prefer me to do it in one of above ways (short of salt - that I won't do), or have some other idea let me know.

File: test/kubemark/start-kubemark.sh (hunk 5 above)
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: Done.

File: test/kubemark/stop-kubemark.sh (hunk 4 above)
Comment (gmarek) [inline test/kubemark/stop-kubemark.sh]: Done.

File: test/kubemark/start-kubemark.sh (hunk 3 above)
Comment (quinton-hoole) [inline test/kubemark/start-kubemark.sh]: I prefer passing it to start-kubemark-master.sh as you suggest. It's not all about basic readability - it's more about reducing coupling. https://en.wikipedia.org/wiki/Coupling_(computer_programming)

File: test/kubemark/start-kubemark-master.sh (hunk 1 above)
Comment (quinton-hoole) [inline test/kubemark/start-kubemark-master.sh]: See comment below.  Rather pass as a command-line argument.

**Comment by quinton-hoole** [comment]
One more minor change and then LGTM.

**Comment by quinton-hoole** [comment]
@gmarek ^^

File: test/kubemark/start-kubemark-master.sh (hunk 1 above)
Comment (gmarek) [inline test/kubemark/start-kubemark-master.sh]: Done.

File: test/kubemark/start-kubemark.sh (hunk 3 above)
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: Done.

**Comment by gmarek** [comment]
@quinton-hoole PTAL

**Comment by quinton-hoole** [comment]
Perfect!  LGTM.

**Comment by gmarek** [comment]
Failures are flakes and this change touches only Kubemark scripts, so I'm going to merge it by hand.

**Pull request merged**